### ISSUE TITLE
New Work : Model Implementation (VIII)

### ISSUE THREAD

**Issue by eeshan15** [The-Red-Wood-Lab/Micro-Gas-Turbine-electrical-energy-Prediction#15]
## Model Description
**Elastic Net Regression**: Elastic Net is a type of regression that is a combination of both Lasso and Ridge regression techniques. It addresses some of the limitations of both techniques and helps in situations where there are multiple correlated independent variables. The Elastic Net penalty is a mix of L1 (Lasso) and L2 (Ridge) regularizations.

So far, we have applied the XGBoost & Multivariate Linear Regression algorithms, followed by Linear Regression as our third model. We will continue experimenting with different models to find the best one.

## PR Submission Details

1. **Download the dataset** (if you haven't). You can find the links in issue number 4: https://github.com/The-Red-Wood-Lab/Micro-Gas-Turbine-electrical-energy-Prediction/issues/4
2. **Apply the Elastic Net Regression model**. Do not worry if you don't know how to apply it; here is an article from where you can learn Elastic Net Regression. Click over this link: [Article](https://www.geeksforgeeks.org/videos/what-is-elastic-net-regression-in-machine-learning/)
3. **Calculate the RMSE and R2 Score**. The Cost Function concept is similar to Multivariate Linear Regression. This article explains it well. Click over this link: [Article](https://www.geeksforgeeks.org/ml-linear-regression/#cost-function-for-linear-regression)
4. You can learn Deeply be learning its mathematics and actual Functioning behind it you can read this [Article](https://akbarikevin.medium.com/understanding-elastic-net-regression-a-brief-guide-with-mathematical-insights-bf7571379303)
5. For a **successful submission of the PR**, take **screenshots of the graphs** that show the relation between the predicted values of the **Voltage column** based on other dependent columns, which are time and voltage. Below is the description:
6. Take the training dataset and split it into a **75:25 ratio** for training and validation datasets. You can also use the **K-Fold Cross-Validation** method.
7. Next, **check for overfitting** by visualizing training and validation loss by plotting them in a single graph. If your validation curve is rising and the training curve is declining, there is an issue that needs addressing. You can use **regularization techniques** to solve the problem.
8. **Show the following scores** in your PR. The needed scores are outlined in the checklist section.

## Checklist

- [ ] Applied Elastic Net Regression
- [ ] Applied Regularization to Fix Overfitting, if existing
- [ ] Attached Screenshots of the Graphs
- [ ] Showcased Your Train, Validation, and Test RMSE and R2 Scores

## Conditions
A group of people will be working on the issue, and the best model will be evaluated based on the lowest RMSE score amongst them and an R2 score close to 1. This creates a leaderboard system. If your model shows signs of overfitting, your PR will be rejected.
